packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/InAppPurchasePlugin.swift

```diff
@@ -0,0 +1,412 @@
+          message:
+            "Failed to get SKProductResponse in startRequest call. Error occurred on iOS platform",
+          details: productIdentifiers)
+        completion(nil, error)
+        return
+      }
+
+      response.products.forEach { product in
+        self.productsCache[product.productIdentifier] = product
+      }
+
+      if #available(iOS 12.2, *) {
+        if let responseMessage = FIAObjectTranslator.convertProductsResponse(toPigeon: response) {
+          completion(responseMessage, nil)
+        }
+      }
+      self.requestHandlers.remove(handler)
+    }
+  }
+
+  public func addPaymentPaymentMap(
+    _ paymentMap: [String: Any], error: AutoreleasingUnsafeMutablePointer<FlutterError?>
+  ) {
+    guard let productID = paymentMap["productIdentifier"] as? String else {
+      error.pointee = FlutterError(
+        code: "storekit_missing_product_identifier",
+        message: "The `productIdentifier` is missing from the payment map.",
+        details: paymentMap)
+      return
+    }
+
+    guard let product = self.getProduct(productID: productID) else {
+      error.pointee = FlutterError(
+        code: "storekit_invalid_payment_object",
+        message:
+          "You have requested a payment for an invalid product. Either the `productIdentifier` of the payment is not valid or the product has not been fetched before adding the payment to the payment queue.",
+        details: paymentMap)
+      return
+    }
+
+    let payment = SKMutablePayment(product: product)
+    payment.applicationUsername = paymentMap["applicationUsername"] as? String
+    payment.quantity = paymentMap["quantity"] as? Int ?? 1
+    payment.simulatesAskToBuyInSandbox = paymentMap["simulatesAskToBuyInSandbox"] as? Bool ?? false
+
+    if #available(iOS 12.2, *) {
+      if let paymentDiscountMap = paymentMap["paymentDiscount"] as? [String: Any],
+        !paymentDiscountMap.isEmpty
+      {
+        var invalidError: NSString?
+        if let paymentDiscount = FIAObjectTranslator.getSKPaymentDiscount(
+          fromMap: paymentDiscountMap, withError: &invalidError)
+        {
+          payment.paymentDiscount = paymentDiscount
+        } else if let invalidError = invalidError {
+          error.pointee = FlutterError(
+            code: "storekit_invalid_payment_discount_object",
+            message:
+              "You have requested a payment and specified a payment discount with invalid properties. \(invalidError)",
+            details: paymentMap)
+          return
+        }
+      }
+    }
+
+    guard self.paymentQueueHandler?.add(payment) == true else {
+      error.pointee = FlutterError(
+        code: "storekit_duplicate_product_object",
+        message:
+          "There is a pending transaction for the same product identifier. Please either wait for it to be finished or finish it manually using `completePurchase` to avoid edge cases.",
+        details: paymentMap)
+      return
+    }
```

Comment:
does it have any more logic after the guard? A guard statement is typically used as an early exit of a function